tests/connect.robot

```diff
@@ -12,7 +13,8 @@
 | *Test Cases*
 | Connect to a broker with default port and client id
 | | ${mqttc}    | Connect   | ${broker.uri}
-| | Should Match Regexp     | ${mqttc._client_id}   | ^paho/[0-9A-f]{18}
+| | ${client_id} = | Decode Bytes To String | ${mqttc._client_id} | UTF-8
+| | Should Be Empty         | ${client_id}  |
```

Comment:
Why should the client_id empty?

Reply:
`client_id` is only generated for protocol version `MQTTv31`: https://github.com/eclipse/paho.mqtt.python/blob/master/src/paho/mqtt/client.py#L541
But protocol defaults to `MQTTv311` in the client constructor: https://github.com/eclipse/paho.mqtt.python/blob/master/src/paho/mqtt/client.py#L493

So either we have to provide a randomly generated `client_id` in MQTTLibrary, or it will be set to an empty byte array.

Reply:
Ok thanks. we already have a test with specific client id, so this is not an issue.